### PR TITLE
Added autoClose prop for Notification and migrated JS tests to TS

### PR DESCRIPTION
**Pull request by g4rry420**
Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>

#### What does this PR do?
Added autoClose prop so that notification toast don't close automatically and user has control over when they want to close a notification toast.
Also, I have converted js tests to tsx.

#### Where should the reviewer start?
components => Notification

#### What testing has been done on this PR?
prewritten tests are passing successfully

#### How should this be manually tested?
Using storybook

#### Any background context you want to provide?
No

#### What are the relevant issues?
#5681, #5599

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
Yes

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/Notification/__tests__/Notification-test.tsx [hunk 1]
```diff
@@ -46,4 +46,16 @@ describe('Notification', () => {
...
+      <Grommet>
+        <Notification
+          toast={{ autoClose: false }}
+          title="title"
+          message="message"
+        />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
```
[the remark this answers is not in the document]
Comment (g4rry420) [reply src/js/components/Notification/__tests__/Notification-test.tsx]: I have enhanced the test but not sure if this is the correct way. Please, let me know about it.

File: src/js/components/Notification/__tests__/Notification-test.tsx [hunk 2]
```diff
@@ -44,4 +45,19 @@ describe('Notification', () => {
     fireEvent.click(screen.getByRole('button'));
     expect(onClose).toBeCalled();
   });
+
+  test('autoClose', async () => {
+    jest.useFakeTimers('modern');
+    const { getAllByText } = render(
+      <Grommet>
+        <Notification
+          toast={{ autoClose: false }}
```
[the remark this answers is not in the document]
Comment (g4rry420) [reply src/js/components/Notification/__tests__/Notification-test.tsx:54]: Do you think that the test is written in correct way ? Coz I am not sure if this is correct way.

[the remark this answers is not in the document]
Comment (jcfilben) [reply src/js/components/Notification/__tests__/Notification-test.tsx:54]: I wrote the following test which does work, but I am getting some interesting behavior where it only passes when I run it by itself. When run with the rest of the tests it fails. Looking into it more...
```javascript
test.only('autoClose true', async () => {
  jest.useFakeTimers('modern');
  render(
    <Grommet theme={grommet}>
      <Notification
        toast={{ autoClose: true }}
        title="test"
        message="message"
      />
    </Grommet>,
  );
  expect(screen.getByText('test')).toBeInTheDocument();
  act(() => {
    jest.advanceTimersByTime(9000);
  });
  expect(screen.queryByText('test')).not.toBeInTheDocument();
});
```

**Review by amandadupell — APPROVED** [review]
besides specific test behavior, this looks good to me!

**Comment by halocline** [comment]
@danielm-hpe - Does this meet your desired functionality?

**Comment by danielm-hpe** [comment]
> @danielm-hpe - Does this meet your desired functionality?

Yes this should work, thanks @g4rry420

File: src/js/components/Notification/__tests__/Notification-test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcfilben) [reply src/js/components/Notification/__tests__/Notification-test.tsx]: I just updated this PR with new tests that are working as expected and test both `autoClose: true` and `autoClose: false` behavior.

**Review by jcfilben — APPROVED** [review]
I updated the tests so we have a test for both `autoClose: true` and `autoClose: false`. I also fixed merged conflicts, but when I merged with master it pulled in some lint changes for `Carousel/propTypes.js` and `Form/Form.js`.

I think this PR is ready to go

**Comment by g4rry420** [comment]
Thank you @jcfilben  for your help in making this pr successful.

**Comment by danielm-hpe** [comment]
Just following up to check on this PR- it looks like it's basically ready other than needing to resolve merge conflicts?

**Review by ericsoderberghp — APPROVED** [review]

**Comment by ericsoderberghp** [comment]
> Just following up to check on this PR- it looks like it's basically ready other than needing to resolve merge conflicts?

agreed

**Review by jcfilben — APPROVED** [review]

**Pull request merged**